src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ISwitchStatement.cs

```diff
@@ -0,0 +1,1213 @@
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<SwitchStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(17600, "https://github.com/dotnet/roslyn/issues/17600")]
+        public void ISwitchStatement_SwitchSectionWithLocalDeclarataion()
+        {
+            string source = @"
+using System;
+
+class Test
+{
+    int M(int number)
+    {
+        /*<bind>*/switch (number)
+        {
+            case 0:
+            case 1:
+                int number2 = 2;
+                return number2;
+        }/*</bind>*/
+
+        return 0;
+    }
+}
+";
+            string expectedOperationTree = @"
+ISwitchStatement (1 cases) (OperationKind.SwitchStatement) (Syntax: 'switch (num ... }')
+  Switch expression: 
+    IParameterReferenceExpression: number (OperationKind.ParameterReferenceExpression, Type: System.Int32) (Syntax: 'number')
+  Sections:
+      ISwitchCase (2 case clauses, 2 statements) (OperationKind.SwitchCase) (Syntax: 'case 0: ... rn number2;')
+          Clauses:
+              ISingleValueCaseClause (CaseKind.SingleValue) (OperationKind.CaseClause) (Syntax: 'case 0:')
+                Value: 
+                  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0) (Syntax: '0')
+              ISingleValueCaseClause (CaseKind.SingleValue) (OperationKind.CaseClause) (Syntax: 'case 1:')
+                Value: 
+                  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1) (Syntax: '1')
+          Body:
+              IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'int number2 = 2;')
+                IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'number2 = 2')
+                  Variables: Local_1: System.Int32 number2
+                  Initializer: 
+                    ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2) (Syntax: '2')
+              IReturnStatement (OperationKind.ReturnStatement) (Syntax: 'return number2;')
+                ReturnedValue: 
+                  ILocalReferenceExpression: number2 (OperationKind.LocalReferenceExpression, Type: System.Int32) (Syntax: 'number2')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<SwitchStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(17600, "https://github.com/dotnet/roslyn/issues/17600")]
+        public void ISwitchStatement_SimpleCaseClauseWithLocalDeclarataion()
+        {
+            string source = @"
+using System;
+
+class Test
+{
+    void M(int number)
+    {
+        /*<bind>*/switch (number)
+        {
+            case 0:
+            case M2(out var number2):
```

Comment:
>M2(out var number2) [](start = 17, length = 19)

Are there tests to request nodes for case expression directly? What about when clause?